sources/Inputs/src/NetworkManager.cpp

```diff
@@ -140,6 +144,34 @@ NetworkManager::buildTree() {
       LOG(debug) << "Node " << bus1->getID() << " connected to " << bus2->getID() << " and " << bus3->getID() << " by 3W " << (*it_t)->getID() << LOG_ENDL;
     }
   }
+
+  const auto& hvdcLines = network->getHvdcLines();
+  for (auto hvdcLine : hvdcLines) {
+    auto converter_dyn_1 = hvdcLine->getConverter1();
+    auto converter_1 = ConverterInterface(converter_dyn_1->getID(), converter_dyn_1->getBusInterface()->getID());
```

Comment:
Maybe check that the buses are connected. @marcochiaramello , could you confirm ?

Reply:
indeed, if converters are not connected, we do not need to specify anything in dyd

Reply:
So, if there are one of the converter of the hvdcLine which is in a bus that is not connected, I must skip this hvdc line in this for loop ?

Reply:
What do you mean by "not connected" @lecourtoisflo ?

Reply:
elemenst (generators, loads, etc...) have an API getInitialConnected which determines if they are connected to the network. Since we are using it for generators, I thought it would be a good idea to use it for converters